TICKET-4:

simyara opened the pull request:
related to ticket [Ticket-4](https://trello.com/c/nTCKT7It/2-04-update-picture)
pull request updated

simyara committed:
TICKET-4:
[+] body parser middleware
[+] object validation method
[+] update single picture method

ikryvanos commented on actions/singlePicture.js:
```diff
@@ -1,9 +1,21 @@
 "use strict";
 
 let pictureServer = require('../services/pictureServer');
+let pictureValidator = require('../middleware/validator');
```
Please, move validator to the services folder

ikryvanos commented on actions/singlePicture.js:
```diff
@@ -1,9 +1,21 @@
 let pictureServer = require('../services/pictureServer');
+let pictureValidator = require('../middleware/validator');
+
+function updateItem(origObj, dataToUpdate){
```
Add lodash as dependencie, and use merge method https://lodash.com/docs/4.17.4#merge

ikryvanos commented on middleware/validator.js:
```diff
@@ -0,0 +1,54 @@
+    });
+}
+
+module.exports = {
```
Make validator like just a service. It shouldn't work with response body

ikryvanos commented on middleware/validator.js:
```diff
@@ -0,0 +1,54 @@
+
+function validate(obj, schema, res){
+    return Object.keys(obj).every(function(element) {
+        if (schema.hasOwnProperty(element)) {
```
use just 
```javascript
if (schema[element])
```
because if it has string value by key it will be truly value

ikryvanos replied:
Try to avoid using `else` where it is possible.
Create variable before `if` with default value
Change it in `if`, 
And return after all

ikryvanos commented on package.json:
```diff
@@ -10,9 +10,11 @@
+    "co-body": "^5.1.1",
     "koa": "~1.2.5",
     "koa-body": "~1.4.0",
+    "koa-bodyparser": "^2.5.0",
```
Remove unnecessary dependency

simyara committed:
TICKET-4:
[*] move validator to the services folder
[+] lodash
[*] refactoring validate function
[-] unnecessary dependency

ikryvanos commented on actions/singlePicture.js:
```diff
@@ -35,6 +25,15 @@ module.exports = {
     },
     updateOnePicture: function*(next) {
+
+        let validationResult = yield validator.validateObject(this.request.body);
```
use 2 parameters: request data and schema

ikryvanos commented on services/validator.js:
```diff
@@ -0,0 +1,57 @@
+"use strict";
+
+const schema = {
```
Remove schema from here

ikryvanos commented on services/validator.js:
```diff
@@ -0,0 +1,57 @@
+
+        if (!schema[element]) {
+            resultObj.isValid = false;
+            resultObj.status = 400;
```
add to result only error message and `isValid` flag

ikryvanos commented on services/validator.js:
```diff
@@ -0,0 +1,57 @@
+        isValid: true
+    };
+
+    Object.keys(obj).every(function(element) {
```
use `reduce` or `map` and `filter` instead of every

ikryvanos commented on services/validator.js:
```diff
@@ -0,0 +1,57 @@
+}
+
+module.exports = {
+    validateObject:  function* (obj){
```
export validate method without wrapping

simyara committed:
TICKET-4:
[*] move validator to the services folder
[+] lodash
[*] refactoring validate function
[-] unnecessary dependency

ikryvanos commented on services/validator.js:
```diff
@@ -1,57 +1,43 @@
     });
-    return resultObj;
+
+    console.log(errorString);
```
Please, remove loging

ikryvanos commented on services/validator.js:
```diff
@@ -1,57 +1,43 @@
+        errorMessage: errorString.join()
+    };
+
+    console.log(resObj);
```
Please, remove loging

simyara committed:
TICKET-4:
[-] logging
